### ISSUE TITLE
Can't provision salt-cloud servers in parallel in 2015.8.3

### ISSUE THREAD

**Issue by dnd** [saltstack/salt#30547]
In 2015.8.3 I am unable to provision servers in parallel. The error I receive when I attempt passing `-P` to salt-cloud is:

```
    ----------
    Provisioning:
        VM being provisioned in parallel. PID: 17022
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1189, in create
    self.opts
  File "/usr/lib/python2.7/dist-packages/salt/utils/cloud.py", line 159, in gen_keys
    salt.crypt.gen_keys(tdir, 'minion', keysize)
  File "/usr/lib/python2.7/dist-packages/salt/crypt.py", line 88, in gen_keys
    gen = RSA.generate(bits=keysize, e=65537)
  File "/usr/lib/python2.7/dist-packages/Crypto/PublicKey/RSA.py", line 508, in generate
    obj = _RSA.generate_py(bits, rf, progress_func, e)    # TODO: Don't use legacy _RSA module
  File "/usr/lib/python2.7/dist-packages/Crypto/PublicKey/_RSA.py", line 50, in generate_py
    p = pubkey.getStrongPrime(bits>>1, obj.e, 1e-12, randfunc)
  File "/usr/lib/python2.7/dist-packages/Crypto/Util/number.py", line 265, in getStrongPrime
    randfunc)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/_UserFriendlyRNG.py", line 202, in read
    return self._singleton.read(bytes)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/_UserFriendlyRNG.py", line 178, in read
    return _UserFriendlyRNG.read(self, bytes)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/_UserFriendlyRNG.py", line 132, in read
    retval = self._fa.random_data(N)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/Fortuna/FortunaAccumulator.py", line 148, in random_data
    return self.generator.pseudo_random_data(bytes)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/Fortuna/FortunaGenerator.py", line 93, in pseudo_random_data
    for i in xrange(num_full_blocks):
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/_UserFriendlyRNG.py", line 202, in read
    return self._singleton.read(bytes)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/_UserFriendlyRNG.py", line 178, in read
    return _UserFriendlyRNG.read(self, bytes)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/_UserFriendlyRNG.py", line 132, in read
    retval = self._fa.random_data(N)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/Fortuna/FortunaAccumulator.py", line 148, in random_data
    return self.generator.pseudo_random_data(bytes)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/Fortuna/FortunaGenerator.py", line 93, in pseudo_random_data
    for i in xrange(num_full_blocks):
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/_UserFriendlyRNG.py", line 202, in read
    return self._singleton.read(bytes)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/_UserFriendlyRNG.py", line 178, in read
    return _UserFriendlyRNG.read(self, bytes)
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/_UserFriendlyRNG.py", line 137, in read
    self._check_pid()
  File "/usr/lib/python2.7/dist-packages/Crypto/Random/_UserFriendlyRNG.py", line 153, in _check_pid
    raise AssertionError("PID check failed. RNG must be re-initialized after fork(). Hint: Try Random.atfork()")
AssertionError: PID check failed. RNG must be re-initialized after fork(). Hint: Try Random.atfork()
```

This is attempting to provision to Linode if it makes any difference.

**Comment by dnd** [comment]
I think this is actually a duplicate of #28396 which looks to be fixed.

**Comment by rallytime** [comment]
@dnd Yes, indeed. That fix will come out in the upcoming 2015.8.4 release.
